### ISSUE TITLE
Structured VQ ResNet Experiments

### ISSUE THREAD

**Issue by FlorianPfisterer** [Simsso/NIPS-2018-Adversarial-Vision-Challenge#75]
### Hyperparameters to Test

| Description | Parameter Name | Interval |
|---|---|---|
| number of embedding vectors | `n` | |
| vq-layer position and number | - | each layer |
| virtual batch size | `batch_size * factor` | 32 - 10,000 |
| vector size | `q / num_splits` | 1 - q |
| alpha, beta, gamma | `alpha, beta, gamma` | ? |
| learning rate | `learning_rate` | 1 |
| distance norm type | `lookup_ord` | 1, 2, inf, cosine |
| replacement number | `num_replacements` | ≤ 10% of batch size |
| initializer stddev, mean | `embedding_initializer` | _according to inputs_ | 

### Goals
* achieve accuracy > 60% with VQ-layer 
* devise a relation between accuracy and robustness
